go-controller/pkg/ovn/hybrid.go

```diff
@@ -180,11 +184,15 @@ func (oc *DefaultNetworkController) setupHybridLRPolicySharedGw(nodeSubnets []*n
 			matchStr := fmt.Sprintf(`inport == "%s%s" && %s.dst == %s`,
 				ovntypes.RouterToSwitchPrefix, nodeName, L3Prefix, hybridCIDR)
 
+			name := ovntypes.HybridSubnetPrefix + nodeName + ":" + hoNodeName
+			if hoNodeName == hybridCidrClusterKey {
+				name = ovntypes.HybridSubnetPrefix + nodeName
+			}
 			// Logic route policy to steer packet from pod to hybrid overlay nodes
 			logicalRouterPolicy := nbdb.LogicalRouterPolicy{
 				Priority: ovntypes.HybridOverlaySubnetPriority,
 				ExternalIDs: map[string]string{
-					"name": ovntypes.HybridSubnetPrefix + nodeName,
+					"name": name,
```

Comment:
what about upgrades? Don't we need a sync upgrade function to convert the old names to the new ones? Or is it not applicable?

I'm a little confused by "In cases where config.HybridOverlay.ClusterSubnets is not provided,", what is this use case? Is it for SDN migration? Might be good to have comments on line 157 and in the else statement indicating the use cases.

Reply:
Yes, this is for live migration. We will reuse the host subnets allocated by OpenShiftSDN as the `hybrid-overlay-node-subnet` and set up hybrid overlay routes/policies to these subnets. So it is not applicable for upgrade. I'll add comments for it.